src/data/firebase.js

```diff
@@ -0,0 +1,95 @@
+import { initializeApp } from 'firebase/app';
+import { getDatabase, ref, get } from 'firebase/database';
+import { firebaseConfig } from '../environment/firebase.env';
+
+initializeApp(firebaseConfig);
+
+function getLocalData(section = null) {
+  const dataFromStorage = JSON.parse(localStorage.getItem('DIMS'));
+  if (dataFromStorage) {
+    return section ? dataFromStorage[section] : dataFromStorage;
+  }
+  localStorage.setItem(
+    'DIMS',
+    JSON.stringify({
+      users: [],
+      tasks: [],
+      tasksProgress: [],
+    }),
+  );
+
+  return getLocalData(section);
+}
+
+function setLocalData(section, data) {
+  const localData = getLocalData();
+  localData[section] = data;
+  localStorage.setItem('DIMS', JSON.stringify(localData));
+}
+
+async function getUsersData() {
+  const localData = getLocalData('users');
+  let unsortData;
+  if (localData.length > 0) {
+    unsortData = localData;
+  } else {
+    const db = getDatabase();
+    const usersListRef = ref(db, 'users');
+    const usersData = await get(usersListRef);
+    if (usersData.exists()) {
+      unsortData = usersData.val();
+    }
+  }
+  setLocalData('users', unsortData);
+
+  return unsortData.filter((item) => item);
+}
+
+async function getUserProgressData(progressId = []) {
+  if (!progressId.length) {
+    return null;
+  }
+  const localData = getLocalData('tasksProgress');
+  let unsortData;
+  if (localData.length) {
+    unsortData = localData;
+  } else {
+    const dbRef = ref(getDatabase(), 'tasksProgress');
+    const progressData = await get(dbRef);
+    unsortData = progressData.val();
+  }
+  setLocalData('tasksProgress', unsortData);
+
+  return unsortData.filter((item) => {
+    if (!item) {
+      return false;
+    }
+
+    return progressId.some((id) => id === item.id);
+  });
+}
+
+async function getUserTasksData(tasksId = []) {
+  if (!tasksId.length > 0) {
```

Comment:
`!tasksId.length`

Reply:
I rewrote this method and now I don't use this method and I don't use double (and more) filter anywhere.
```suggestion
async function getUserTasks() {
  const querySnapshot = await getDocs(collection(db, 'userTasks'));
  if (!querySnapshot.empty) {
    const userTasksData = querySnapshot.docs.map((document) => {
      return {
        ...document.data(),
        id: document.id,
      };
    });
    return userTasksData;
  }
  return null;
}
```